File: Python/phy_che_mat_results_passed_calc.py
class results:
    def __init__(self, a=0, b=0, c=0):
        self.__phy = a
        self.__che = b
        self.__mat = c
        self.__passcount = 0
        self.__defphy(self.__phy)
        self.__defphy(self.__che)
        self.__defphy(self.__mat)
        self.__calc()
        self.__display()
    def __defphy(self, phy):
        if phy > 0 and phy < 101:
            phy = phy
        else:
            print("phy invalid")
        if phy > 69:
            self.__passcount = self.__passcount + 1
    def __defche(self, che):
        if che > 0 and che < 101:
            che = che
        else:
            print("che invalid")
        if che > 69:
            self.__passcount = self.__passcount + 1
    def __defmat(self, mat):
        if mat > 0 and mat < 101:
            mat = mat
        else:
            print("mat invalid")
        if mat > 69:
            self.__passcount = self.__passcount + 1
    def __calc(self):
        self.__total = self.__phy + self.__che + self.__mat
        self.__per = self.__total * 100 / 300
    def __display(self):
        print(self.__total)
        print(self.__per)
        if self.__passcount == 0:
            print("go home")
        if self.__passcount == 1:
            print("retake course")
        if self.__passcount == 2:
            print("retake exam")

richmond = results(69, 69, 69)

File: Java/PhyCheMatResultsPassedCalc.java

public class PhyCheMatResultsPassedCalc
{	
	private int phy, che, mat, passed;
	private float total, per;	
	public void display()
	{
		if (phy == -1 || che == -1 || mat == -1)
			System.out.println("Invalid");
		else
		{
			total = phy + che + mat;
			per = total * 100 / 300;
			System.out.println("Total: " + total);
			System.out.println("Per: " + per);
			System.out.println("Passed: " + passed);
			if (passed == 0)
				System.out.println("Go home");
			if (passed == 1)
				System.out.println("Retake course");
			if (passed == 2)
				System.out.println("Retake exam");
		}
	}
	public void physics(int a)
	{
		che = -1;
		if (a >= 0 & a <= 100)
			che = a;
		else
			System.out.println("Physics Invalid");
		if (che >= 70)
			passed = passed + 1;
	}
	public void chemistry(int a)
	{
		phy = -1;
		if (a >= 0 & a <= 100)
			phy = a;
		else
			System.out.println("Chemistry Invalid");
		if (phy >= 70)
			passed = passed + 1;
	}
	public void mathematics(int a)
	{
		mat = -1;
		if (a >= 0 & a <= 100)
			mat = a;
		else
			System.out.println("Mathematics Invalid");
		if (mat >= 70)
			passed = passed + 1;		
	}
}
